great_expectations/agent/actions/list_table_names.py

```diff
@@ -0,0 +1,38 @@
+from typing import TYPE_CHECKING
+
+from great_expectations.agent.actions.agent_action import (
+    ActionDetails,
+    ActionResult,
+    AgentAction,
+)
+from great_expectations.agent.models import (
+    ListTableNamesEvent,
+)
+from great_expectations.compatibility.sqlalchemy import inspect
+from great_expectations.datasource.fluent import Datasource as FluentDatasource
+from great_expectations.datasource.fluent import SQLDatasource
+
+if TYPE_CHECKING:
+    from great_expectations.compatibility.sqlalchemy.engine import Inspector
+
+
+class ListTableNamesAction(AgentAction[ListTableNamesEvent]):
+    def run(self, event: ListTableNamesEvent, id: str) -> ActionResult:
+        datasource_name: str = event.datasource_name
+        datasource: FluentDatasource = self._context.get_datasource(
+            datasource_name=datasource_name
+        )
+        if not isinstance(datasource, SQLDatasource):
+            raise TypeError(
+                f"This operation requires a SQL Datasource but got {type(datasource).__name__}."
+            )
+
+        inspector: Inspector = inspect(datasource.get_engine())
+        table_names: list[str] = inspector.get_table_names()
```

Comment:
```suggestion
        inspector: Inspector = inspect(datasource.get_engine())
        table_names: List[str] = inspector.get_table_names()
```
Need to use `from typing import List` here

Reply:
Good catch!